From 1fe5eec13e1e9c1edc2e7ad687566b549dbb9f80 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 20 Sep 2018 08:19:36 -0600
Subject: [PATCH 5/5] klte-common: Set SDK API level for mediaserver and
 mm-qcamera-daemon

* This singularly handles both text relocation (M) and mutex (O)
  linker restrictions
* Due to blobs, both need to enable text relocations and mm-qcamera-daemon
  needs the mutex restrictions relaxed. API 22 is the lowest common
  denominator to allow both.

Change-Id: I97a79fb57aaadf5cf62894489522b2de6d8c2e36
---
 BoardConfigCommon.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 71ea933..16591b4 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -52,9 +52,11 @@ LZMA_RAMDISK_TARGETS := recovery
 TARGET_KERNEL_SOURCE := kernel/samsung/msm8974
 
 # Legacy BLOB Support
-TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS := true
 TARGET_LD_SHIM_LIBS += \
     /system/vendor/lib/hw/camera.vendor.msm8974.so|libshim_camera.so
+TARGET_PROCESS_SDK_VERSION_OVERRIDE += \
+    /system/bin/mediaserver=22 \
+    /system/vendor/bin/mm-qcamera-daemon=22
 
 # Partitions
 BOARD_BOOTIMAGE_PARTITION_SIZE := 13631488
-- 
2.17.1

